### ISSUE TITLE
BIM-->PBIX Memory Failure

### ISSUE THREAD

**Issue by dstanleyathighlinestoragepartnerscom** [TabularEditor/TabularEditor#1210]
I'm following the TE2 documentation workaround to create a PBIX file from my BIM file now that I have enabled the xmla endpoint write capabilities. I'm following the steps and continuing to get a memory error when I open the PBIP file:

![image](https://github.com/TabularEditor/TabularEditor/assets/174831296/073f01fd-5725-4ace-b519-964300865e6c)

Furthermore, when I then save as a PBIX and refresh even one table I'm getting a memory error msg.

I'll note that I'm able to see the data tables when I select edit query from the PBIX file to display Power Query, but not from the actual data table view in PowerBI. Thanks for any advice or recommendation to ultimately be able to create a PBIX file since I'm unable to DL from PowerBI Online.

**Comment by otykier** [comment]
Which workaround did you use?

**Comment by otykier** [comment]
I don't believe it's possible to replace the .bim file within a .pbit (Power BI Template) file.

You should use PBIP (Power BI Project) folder format instead, as described here: https://docs.tabulareditor.com/te3/powerbi-xmla-pbix-workaround.html

More info about how to enable PBIP / Developer Mode: https://learn.microsoft.com/en-us/power-bi/developer/projects/projects-overview

**Comment by dshsp** [comment]
I’m sorry I misspoke, that’s what I did with a pbip formatted file using that exact tutorial. Are there any other workarounds? Thanks again for your help.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by otykier** [comment]
I see.

Once you replace the model.bim file, but before you open the PBIP folder in Power BI Desktop, could you try to delete the following files from the PBIP folder structure:

.pbi\localSettings.json
.pbi\editorSettings.json
.pbi\cache.abf
.pbi\unapplicedChanges.json

More information about these files here: https://learn.microsoft.com/en-us/power-bi/developer/projects/projects-dataset

**Comment by dshsp** [comment]
Unfortunately, I’m continuing to get the below error when I load the pbip file. Otherwise, if I edit directly from tabular editor and update via the powerbi online interface everything is perfect…I just can’t get back to having a pbix copy, which I understand was a risk. Thanks for continuing to help with this, very much appreciated. This tool is fantastic.

***@***.***

**Comment by otykier** [comment]
Hmm, that's unfortunate. Even though Developer Mode (PBIP) is still in preview, making changes to the Model.bim file or replacing the Model.bim file as you did, should be fully supported.

@ruiromano, have you seen this issue before? Got any advice?

**Comment by RuiRomano** [comment]
Yes, replacing the model.bim is supported. If the external changes applied respect the modelling authoring list: https://learn.microsoft.com/en-gb/power-bi/developer/projects/projects-overview#model-authoring

@dshsp you get the error when you open the PBIP? When you refresh? Do you have any report visuals? Can you try to open the PBIP with an empty report (just delete the report.json file)? 

I also suggest you monitor PBI Desktop memory while opening using task manager.

**Comment by dshsp** [comment]
Thanks for the reply. I get the error when I open the PBIP file after replacing the .bim file within the semantic structure. After the error message the file continues to open and it looks normal, with all of the data tables, measures, etc visible within the Data pane on the right, however, there is no data that has been loaded behind it (I can even click on an individual measure and see the DAX code). I can even see the relationships in the model view. It’s odd because when I click on an individual table from the Data pane and click edit query to view within the powerquery window, I can see all of the data for every table just fine, even after hitting refresh preview for all. However, after clicking close and apply the refresh in the file throws a memory error once again.

The report is blank right now, no visuals. I tried the deletion suggestion below for the report.json file and unfortunately didn’t even load the file as it said the report.json file was missing. Thank you for any follow ups….
